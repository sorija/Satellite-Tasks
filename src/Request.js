import React from "react";

const Request = ({ id, satName, start, end, status }) => (
  <React.Fragment>
    <ul>
      <p>{id}</p>
      <p>{satName}</p>
      <p>{start}</p>
      <p>{end}</p>
      <p>{status}</p>
    </ul>
  </React.Fragment>
);
export default Request;
