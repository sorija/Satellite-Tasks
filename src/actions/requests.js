export const addRequest = (satName, longitude, latitude) => ({
  type: "ADD_NEW_REQUEST",
  satName,
  longitude,
  latitude
});

export const toggleDone = id => ({
  type: "TOGGLE_DONE",
  id
});
