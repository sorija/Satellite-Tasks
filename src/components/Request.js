import React from "react";

const Request = ({ id, satName, longitude, latitude, done, onChange }) => (
  <div className="request-container">
    <div className="name-and-id">
      <p className="request-id data">
        ID: <span className="data-span">{id}</span>
      </p>
      <p className="sat-name">{satName}</p>
    </div>
    <div className="coordinates">
      <p className="lat data">
        Latitude: <span className="data-span">{latitude}</span>
      </p>
      <p className="lon data">
        Longitude: <span className="data-span">{longitude}</span>
      </p>
    </div>
    <input
      className="request-status data"
      type="checkbox"
      value={id}
      checked={done}
      onChange={onChange}
    />
  </div>
);

export default Request;
