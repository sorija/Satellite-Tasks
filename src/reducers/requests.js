const requestsDefaults = [
  {
    id: 0,
    satName: "Alderan",
    longitude: 4934.539,
    latitude: 34943.023,
    done: false
  },
  {
    id: 1,
    satName: "Bespin",
    longitude: 249809.342,
    latitude: 98342.247,
    done: false
  },
  {
    id: 2,
    satName: "Coruscant",
    longitude: 23634.453,
    latitude: 18374.909,
    done: false
  },
  {
    id: 3,
    satName: "Dagobah",
    longitude: 69529.348,
    latitude: 463290.236,
    done: false
  },
  {
    id: 4,
    satName: "Dagobah",
    longitude: 132487.843,
    latitude: 876341.432,
    done: false
  },
  {
    id: 5,
    satName: "Hoth",
    longitude: 12843.394,
    latitude: 423673.634,
    done: false
  },
  {
    id: 6,
    satName: "Kashyyyk",
    longitude: 56932.893,
    latitude: 484893.208,
    done: false
  },
  {
    id: 7,
    satName: "Naboo",
    longitude: 789644.873,
    latitude: 125423.026,
    done: false
  },
  {
    id: 8,
    satName: "Naboo",
    longitude: 402834.073,
    latitude: 984283.056,
    done: false
  },
  {
    id: 9,
    satName: "Naboo",
    longitude: 673429.233,
    latitude: 3467211.882,
    done: false
  },
  {
    id: 10,
    satName: "Tatooine",
    longitude: 356983.348,
    latitude: 153389.092,
    done: false
  },
  {
    id: 11,
    satName: "Yavin",
    longitude: 934732.342,
    latitude: 175834.037,
    done: false
  },
  {
    id: 12,
    satName: "Wobani",
    longitude: 754094.346,
    latitude: 5645343.056,
    done: false
  }
];

const requestsReducer = (state = requestsDefaults, action) => {
  switch (action.type) {
    case "ADD_NEW_REQUEST":
      return [
        ...state,
        {
          id: state.length,
          satName: action.satName,
          longitude: action.longitude,
          latitude: action.latitude,
          done: false
        }
      ];
    case "TOGGLE_DONE":
      return state.map(request => {
        if (request.id === action.id) {
          return {
            ...request,
            done: !request.done
          };
        } else {
          return request;
        }
      });
    default:
      return state;
  }
};

export default requestsReducer;
