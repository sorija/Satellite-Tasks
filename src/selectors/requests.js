const getDividedRequests = (requests, currentPage) => {
  let startIndex = (currentPage - 1) * 5;
  let endIndex = currentPage * 5;
  return requests.slice(startIndex, endIndex);
};

export default getDividedRequests;
