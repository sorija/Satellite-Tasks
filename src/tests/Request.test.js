import React from "react";
import { shallow } from "enzyme";
import requests from "./fixtures/requests";
import Request from "../components/Request";

test("renders Request component with a request", () => {
  const wrapper = shallow(<Request {...requests[0]} />);
  expect(wrapper).toMatchSnapshot();
});
