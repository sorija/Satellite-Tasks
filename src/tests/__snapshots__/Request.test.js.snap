// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Request component with a request 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Request
    done={false}
    id={0}
    latitude={22222.22}
    longitude={111.11}
    satName="Name0"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="name-and-id"
        >
          <p
            className="request-id data"
          >
            ID: 
            <span
              className="data-span"
            >
              0
            </span>
          </p>
          <p
            className="sat-name"
          >
            Name0
          </p>
        </div>,
        <div
          className="coordinates"
        >
          <p
            className="lat data"
          >
            Latitude: 
            <span
              className="data-span"
            >
              22222.22
            </span>
          </p>
          <p
            className="lon data"
          >
            Longitude: 
            <span
              className="data-span"
            >
              111.11
            </span>
          </p>
        </div>,
        <input
          checked={false}
          className="request-status data"
          type="checkbox"
          value={0}
        />,
      ],
      "className": "request-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="request-id data"
            >
              ID: 
              <span
                className="data-span"
              >
                0
              </span>
            </p>,
            <p
              className="sat-name"
            >
              Name0
            </p>,
          ],
          "className": "name-and-id",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "ID: ",
                <span
                  className="data-span"
                >
                  0
                </span>,
              ],
              "className": "request-id data",
            },
            "ref": null,
            "rendered": Array [
              "ID: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "data-span",
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name0",
              "className": "sat-name",
            },
            "ref": null,
            "rendered": "Name0",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="lat data"
            >
              Latitude: 
              <span
                className="data-span"
              >
                22222.22
              </span>
            </p>,
            <p
              className="lon data"
            >
              Longitude: 
              <span
                className="data-span"
              >
                111.11
              </span>
            </p>,
          ],
          "className": "coordinates",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Latitude: ",
                <span
                  className="data-span"
                >
                  22222.22
                </span>,
              ],
              "className": "lat data",
            },
            "ref": null,
            "rendered": Array [
              "Latitude: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 22222.22,
                  "className": "data-span",
                },
                "ref": null,
                "rendered": 22222.22,
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Longitude: ",
                <span
                  className="data-span"
                >
                  111.11
                </span>,
              ],
              "className": "lon data",
            },
            "ref": null,
            "rendered": Array [
              "Longitude: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 111.11,
                  "className": "data-span",
                },
                "ref": null,
                "rendered": 111.11,
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "className": "request-status data",
          "onChange": undefined,
          "type": "checkbox",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="name-and-id"
          >
            <p
              className="request-id data"
            >
              ID: 
              <span
                className="data-span"
              >
                0
              </span>
            </p>
            <p
              className="sat-name"
            >
              Name0
            </p>
          </div>,
          <div
            className="coordinates"
          >
            <p
              className="lat data"
            >
              Latitude: 
              <span
                className="data-span"
              >
                22222.22
              </span>
            </p>
            <p
              className="lon data"
            >
              Longitude: 
              <span
                className="data-span"
              >
                111.11
              </span>
            </p>
          </div>,
          <input
            checked={false}
            className="request-status data"
            type="checkbox"
            value={0}
          />,
        ],
        "className": "request-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="request-id data"
              >
                ID: 
                <span
                  className="data-span"
                >
                  0
                </span>
              </p>,
              <p
                className="sat-name"
              >
                Name0
              </p>,
            ],
            "className": "name-and-id",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "ID: ",
                  <span
                    className="data-span"
                  >
                    0
                  </span>,
                ],
                "className": "request-id data",
              },
              "ref": null,
              "rendered": Array [
                "ID: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "data-span",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name0",
                "className": "sat-name",
              },
              "ref": null,
              "rendered": "Name0",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="lat data"
              >
                Latitude: 
                <span
                  className="data-span"
                >
                  22222.22
                </span>
              </p>,
              <p
                className="lon data"
              >
                Longitude: 
                <span
                  className="data-span"
                >
                  111.11
                </span>
              </p>,
            ],
            "className": "coordinates",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Latitude: ",
                  <span
                    className="data-span"
                  >
                    22222.22
                  </span>,
                ],
                "className": "lat data",
              },
              "ref": null,
              "rendered": Array [
                "Latitude: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 22222.22,
                    "className": "data-span",
                  },
                  "ref": null,
                  "rendered": 22222.22,
                  "type": "span",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Longitude: ",
                  <span
                    className="data-span"
                  >
                    111.11
                  </span>,
                ],
                "className": "lon data",
              },
              "ref": null,
              "rendered": Array [
                "Longitude: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 111.11,
                    "className": "data-span",
                  },
                  "ref": null,
                  "rendered": 111.11,
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "request-status data",
            "onChange": undefined,
            "type": "checkbox",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
